### PR TITLE
Set line and word direction based on script

### PR DESCRIPTION
**Pull request by mehdisadeghi**
This PR sets word and line direction based on Unicode script of the line. It makes it possible to mix differnt scripts in games text.

### CODE REVIEW

File: src/graphics.c [hunk 1]
```diff
@@ -3048,6 +3053,39 @@ struct word *word_new(const char *str)
...
+	uint32_t dest = 0;
+	uint32_t first[2];
+	u8_toucs(first, 2, str, u8_seqlen(str));
+	int index = 0;
+	/*	Find the first alphanumeric utf8 character for a meaningful direction
+		or use direction of the first character.
+	*/
+	for (int i=0; i < u8len; i++) {
+		dest = u8_nextchar(str, &index);
+		if (!g_unichar_isalnum(dest))
```
Comment (gl00my) [inline src/graphics.c:3068]: style... should be if () {

File: src/graphics.c [hunk 2]
```diff
@@ -5431,6 +5469,25 @@ void _txt_layout_add(layout_t lay, char *txt)
...
+	/* Set direction of each line based on the first non-image, non-number word in that line. */
+	struct word *word = NULL;
+	struct line *ln = NULL;
+	for (ln = layout->lines; ln; ln = ln->next) {
+		for (word = ln->words; word; word = word->next ) {
+			/* Continue until we get a word with some text (not letters or symbols) */
+			if (!word->isalpha)
+				continue;
+
+			if(!word->img) {
```
Comment (gl00my) [inline src/graphics.c:5483]: style: if (), not if()

File: src/graphics.c [hunk 3]
```diff
@@ -3048,6 +3053,39 @@ struct word *word_new(const char *str)
...
+	#ifdef _USE_HARFBUZZ
+	int u8len = u8_strlen(str);
+	uint32_t dest = 0;
+	uint32_t first[2];
+	u8_toucs(first, 2, str, u8_seqlen(str));
+	int index = 0;
+	/*	Find the first alphanumeric utf8 character for a meaningful direction
+		or use direction of the first character.
+	*/
+	for (int i=0; i < u8len; i++) {
```
Comment (gl00my) [inline src/graphics.c:3066]: C99 features are not allowed in INSTEAD.

**Comment by mehdisadeghi** [comment]
@gl00my Thanks for the quick review. Please see my changes!

File: src/graphics.c [hunk 4]
```diff
@@ -3048,6 +3053,39 @@ struct word *word_new(const char *str)
...
+	int index = 0;
+	/*	Find the first alphanumeric utf8 character for a meaningful direction
+		or use direction of the first character.
+	*/
+	for (i=0; i<u8len; i++) {
+		dest = u8_nextchar(str, &index);
+		if (!g_unichar_isalnum(dest)) {
+				dest = 0;
+				continue;
+			}
```
Comment (gl00my) [inline src/graphics.c:3072]: broken indent here }

File: src/graphics.c [hunk 5]
```diff
@@ -3048,6 +3053,39 @@ struct word *word_new(const char *str)
 	w->unbrake = 0;
 	w->prerend = NULL;
 	w->hlprerend = NULL;
+
+	#ifdef _USE_HARFBUZZ
```
Comment (gl00my) [inline src/graphics.c:3057]: It's like that it will be much cleaner to make static function named somehow detect_direction() and move logic there.

**Pull request merged**